src/auth.js

```diff
@@ -1,43 +1,32 @@
-import { storage } from './firebase'
 import { toast } from 'react-toastify'
-import { ref, uploadBytesResumable, getDownloadURL } from 'firebase/storage'
+
 import axios from 'axios'
 
 // REGISTER USER
 export const register = async (email, password, username, file, passwordConfirm, navigate, setShowRegister, setUsername, setEmail, setPassword, setFile, setPasswordConfirm) => {
-  try {
-    const date = new Date().getTime()
-    const storageRef = ref(storage, `${username + date}`)
-    if (passwordConfirm !== password) toast.error('Passwords must match')
+  if (passwordConfirm !== password) toast.error('Passwords must match')
 
-    await uploadBytesResumable(storageRef, file).then(() => {
-      getDownloadURL(storageRef).then(async (downloadURL) => {
-        try {
-          const user = await axios({
-            method: 'POST',
-            url: `${process.env.REACT_APP_API_BASE_URL}/user/register`,
-            data: {
-              username,
-              email,
-              password,
-              photoURL: downloadURL
-            }
-          })
-          navigate('/chat')
-          setShowRegister(false)
-          setUsername('')
-          setEmail('')
-          setPassword('')
-          setFile(null)
-          setPasswordConfirm('')
-          localStorage.setItem('token', user.data.token)
-        } catch (err) {
-          toast.error(err.response.data.message)
-        }
-      })
+  try {
+    const user = await axios.post(`${process.env.REACT_APP_API_BASE_URL}/user/register`, {
+      username,
+      email,
+      password,
+      file
+    }, {
+      headers: {
+        'Content-type': 'multipart/form-data'
+      }
     })
+    navigate('/chat')
+    setShowRegister(false)
+    setUsername('')
+    setEmail('')
+    setPassword('')
+    setFile(null)
+    setPasswordConfirm('')
+    localStorage.setItem('token', user.data.token)
   } catch (err) {
-    console.log(err)
+    toast.error(err.response.data.message)
```

Comment:
Add a fallback here, there are cases where the server might not return a proper response.